### PR TITLE
MCR-2522 allow to limit user selection to particular name identifier type

### PR DESCRIPTION
**Pull request by chemicus**
[Link to jira](https://mycore.atlassian.net/browse/MCR-2522).

### CODE REVIEW

**Review by fluetze — COMMENTED** [review]
Could you please explain, here or in the Jira ticket, why one might wish to limit a match to only certain name identifier types? If so, wouldn't it be better to be able to specify more than one type, and not just one or all?

**Comment by chemicus** [comment]
In the current implementation all nameIdentifiers are used, regardless of being verified or not. The ORCID guidelines require, however, that users need to approve that a work that has been assigned to them, really "belongs" to them. Therefore we should only sync publications where the mods:nameIdentifier[@type='orcid'] has been set, thus the need to limit the type(s) of identifier(s).
I do not see a necessity to limit to more than one identifier at the moment, although one could think about that of course. A general solution would be more complicated, however, since the boolean operator (and, or, ...) between different identifiers would need to be specified as well.
I would therefore suggest to accept the current modification for now, since this is already an improvement. If we see a need for a more general solution, I can certainly add that later.

**Review by fluetze — COMMENTED** [review]
Refactoring ok?

**Comment by chemicus** [comment]
Look ok to me!

**Review by fluetze — DISMISSED** [review]

File: mycore-orcid/src/main/java/org/mycore/orcid/user/MCRORCIDUser.java [hunk 1]
```diff
@@ -134,7 +138,12 @@ public boolean isMyPublication(MCRObjectID oid) {
     public static Set<String> getNameIdentifierKeys(MCRMODSWrapper wrapper) {
         Set<String> identifierKeys = new HashSet<>();
 
-        List<Element> nameIdentifiers = wrapper.getElements("mods:name/mods:nameIdentifier");
+        String xPath = "mods:name/mods:nameIdentifier";
+        if (! MATCH_ONLY_NAME_IDENTIFIER.isEmpty()) {
+            xPath += "[@type = '" + MATCH_ONLY_NAME_IDENTIFIER + "']";
+        }
+
+        List<Element> nameIdentifiers = wrapper.getElements(xPath));
```
Comment (rsteph-de) [inline mycore-orcid/src/main/java/org/mycore/orcid/user/MCRORCIDUser.java]: Syntax error - because of doubled ")" at the end !!!

**Review by rsteph-de — DISMISSED** [review]
Build failed!

**Review by fluetze — DISMISSED** [review]

**Review by kkrebs — APPROVED** [review]

**Pull request merged**